host_core/lib/host_core/host.ex

```diff
@@ -198,7 +198,11 @@ defmodule HostCore.Host do
     credsmap =
       credsmap
       |> Enum.filter(fn {_k, v} ->
-        Map.has_key?(v, "username") || Map.has_key?(v, "password") || Map.has_key?(v, "token")
+        Map.has_key?(v, "registryType") &&
+          (Map.has_key?(v, "username") || Map.has_key?(v, "password") || Map.has_key?(v, "token"))
+      end)
+      |> Enum.map(fn {k, v} ->
+        {extract_server(v["registryType"] |> String.to_existing_atom(), k), v}
```

Comment:
nit: The rest of this statement uses the `Module.func` syntax here, might be easier to read if the Map get here is `Map.get(v, "registryType", nil)`